packages/edit-site/src/components/sidebar-global-styles-wrapper/index.js

```diff
@@ -15,35 +15,43 @@ import { seen } from '@wordpress/icons';
 import GlobalStylesUI from '../global-styles/ui';
 import Page from '../page';
 import { unlock } from '../../lock-unlock';
-import StyleBook from '../style-book';
-import { STYLE_BOOK_COLOR_GROUPS } from '../style-book/constants';
 
 const { useLocation, useHistory } = unlock( routerPrivateApis );
 
 const GlobalStylesPageActions = ( {
 	isStyleBookOpened,
 	setIsStyleBookOpened,
+	path,
 } ) => {
+	const stylebookPath = addQueryArgs( path, { preview: 'stylebook' } );
+	const history = useHistory();
 	return (
 		<Button
 			isPressed={ isStyleBookOpened }
 			icon={ seen }
 			label={ __( 'Style Book' ) }
 			onClick={ () => {
 				setIsStyleBookOpened( ! isStyleBookOpened );
+				const updatedPath = ! isStyleBookOpened
+					? path.replace( '/styles', stylebookPath )
+					: path.replace( stylebookPath, '/styles' );
```

Comment:
I think these replacements aren't quite right, as they don't take into account any subroutes or params of `/styles`, e.g., `/styles?section=%2Fvariations`

To test, navigation to styles > browse styles, then activate the style book. Try to toggle it off or navigation back to the main menu.


https://github.com/user-attachments/assets/72177c4c-50ed-4674-a037-18a2d0f4871a



Would adding and removing the preview query arg be enough?

```js
				const updatedPath = ! isStyleBookOpened
					? stylebookPath
					: removeQueryArgs( path, 'preview' );
```

Reply:
Oooh maybe, let me try!

Reply:
Yes it worked 😄 thanks!